src/ai/ai_controller.c

```diff
@@ -21,14 +21,18 @@
 
 #include "ai_controller.h"
 
+#include <string.h>
+
 #include "main/rom.h"
 #include "memory/memory.h"
+#include "r4300/cp0.h"
+#include "r4300/interupt.h"
+#include "r4300/mi_controller.h"
```

Comment:
These includes were omitted on purpose because they are provided by the r4300_core which acts as the "interface" of the r4300 subsystem for other subsystems (like here the AI). I used that logic during all my refactorings of the different subsystems, that's why there are so many "missing includes".

I can understand that "good style" would be to always include headers for functions that are used in a module, but here I consider these headers to be an implementation detail and therefore not required for inclusion. 

Should I annotate all occurrences of headers that I feel should not be re-added ? Basically {ai,pi,ri,si,vi}_controller.h, {r4300,rdp,rsp}_core.h act as the interface header for their respective subsystems and therefore any other header should not be included in external subsystems.